__tests__/shutdown.test.js

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env node
+/* eslint-env jest */
+
+const fs = require('fs')
+const {promisify} = require('util')
+const Bot = require('../index.js')
+const config = require('./tests.config.js')
+const alice = new Bot()
+const exec = require('child_process').exec
+
+async function doesFileOrDirectoryExist(fpath) {
+  try {
+    await promisify(fs.lstat)(fpath)
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+async function countProcessesMentioning(substr) {
+  expect(substr).toMatch(/^[0-9a-z_\- /]+$/i)
+  const aexec = promisify(exec)
+  try {
+    const execRes = await aexec(`ps ax | grep -v 'grep' | grep "${substr}"`)
+    return execRes.stdout.split('\n').length - 1
+  } catch (e) {
+    if (e.code === 1) {
+      return 0
+    } else {
+      throw new Error('Error looking for processes')
+    }
+  }
+}
+
+async function runTest() {
+  it(`alice can init and deinit()`, async () => {
```

Comment:
Is there a reason for wrapping the test in the `runTest` function?

Reply:
yeah, it's nice to have the test be a series of `await` imperative-looking function calls. but you can't do that except inside an `async` function, I believe. So I had to move it inside a wrapper function like that.

Reply:
oh actually, in this case, the async declaration on the fn passed to `it` suffices so it doesn't need a runTest wrapper... I'm indifferent really. one one hand can drop the wrapper, on the other, we might like tests to look alike and some of them may need it.